rustworkx-core/src/centrality.rs

```diff
@@ -1143,3 +1132,420 @@ where
     }
     closeness
 }
+
+/// Compute the weighted closeness centrality of each node in the graph.
+///
+/// The weighted closeness centrality is an extension of the standard closeness
+/// centrality measure where edge weights represent connection strength rather
+/// than distance. To properly compute shortest paths, weights are inverted
+/// so that stronger connections correspond to shorter effective distances.
+/// The algorithm follows the method described by Newman (2001) in analyzing
+/// weighted graphs.[^Newman]
+///
+/// The edges originally represent connection strength between nodes.
+/// The idea is that if two nodes have a strong connection, the computed
+/// distance between them should be small (shorter), and vice versa.
+/// Note that this assume that the graph is modelling a measure of
+/// connection strength (e.g. trust, collaboration, or similarity).
+/// If the graph is not modelling a measure of connection strength,
+/// the function `weight_fn` should invert the weights before calling this
+/// function, if not it is considered as a logical error.
+///
+/// In the case of a graphs with more than one connected component there is
+/// an alternative improved formula that calculates the closeness centrality
+/// as "a ratio of the fraction of actors in the group who are reachable, to
+/// the average distance".[^WF]
+/// You can enable this by setting `wf_improved` to `true`.
+///
+/// [^Newman]: Newman, M. E. J. (2001). Scientific collaboration networks.
+///     II. Shortest paths, weighted networks, and centrality.
+///     Physical Review E, 64(1), 016132.
+///
+/// [^WF]: Wasserman, S., & Faust, K. (1994). Social Network Analysis:
+///     Methods and Applications (Structural Analysis in the Social Sciences).
+///     Cambridge: Cambridge University Press.
+///     <https://doi.org/10.1017/CBO9780511815478>
+///
+/// # Arguments
+/// * `graph` - The graph object to run the algorithm on
+/// * `wf_improved` - If `true`, scale by the fraction of nodes reachable.
+/// * `weight_fn` - An input callable that will be passed the
+///     `ReversedEdgeReference<<G as IntoEdgeReferences>::EdgeRef>` for
+///     an edge in the graph and is expected to return a `f64` of
+///     the weight of that edge.
+///
+/// # Example
+///
+/// ```rust
+/// use rustworkx_core::petgraph;
+/// use rustworkx_core::centrality::newman_weighted_closeness_centrality;
+/// use crate::rustworkx_core::petgraph::visit::EdgeRef;
+///
+/// // Calculate the closeness centrality of Graph
+/// let g = petgraph::graph::UnGraph::<i32, f64>::from_edges(&[
+///     (0, 1, 0.7), (1, 2, 0.2), (2, 3, 0.5),
+/// ]);
+/// let output = newman_weighted_closeness_centrality(&g, false, |x| *x.weight());
+/// assert_eq!(
```

Comment:
The precision issues also aplies to doctests. So I suggest asserting that a node ismore central than another to keep things simple e.g.

```
assert!(output[1] > output[3])
```